base_layer/core/src/mining/blake_miner.rs

```diff
@@ -74,6 +74,8 @@ impl CpuBlakePow {
                 info!(target: LOG_TARGET, "Mining hash rate per thread: {:.6} MH/s", hash_rate);
                 last_measured_nonce = nonce;
                 start = Instant::now();
+
+                header.timestamp = EpochTime::now();
```

Comment:
While this should be fine, there is a statistical chance this will cause the miner to never find a block.

Reply:
I don't think so. That's not how mining works. You never progress towards mining a block. It is the same as changing the nonce